Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -25,7 +26,7 @@ namespace AzToolsFramework
             auto subWidget = layoutItem->widget();
             if (subWidget)
             {
-                delete subWidget;
+                subWidget->deleteLater();
```

Comment:
I rewrote some of this destruction in my newest PR. I suspect you don't need this anymore. Honestly, I don't think we really need DestroyLayoutContents anymore.

Reply:
I'm just doing deleteLater for safety - handlers were the crash issue I was running into, but any case where you can trigger a layout modification from a widget (which containers might ultimately allow with PropertyRowWidget) has crash potential, so I deferred all deletes across the board by a tick. We can definitely revisit this if it causes a problem / we decide to debounce the actual patch application instead.